### PR TITLE
Add chan_sip on extra port 52xx for dahua and other doorbell support

### PR DESCRIPTION
**Pull request by bdherouville**

### CODE REVIEW

**Comment by felipecrs** [comment]
What's the reason for puttin chan_sip in a different port? Is there any chance it can be used at the same time with pjsip? Example: pjsip for the hass-sip-card and chan_sip for the dahua

**Comment by felipecrs** [comment]
But my point is: if both can't be enabled at the same time, then there's no need to customize their ports either.

**Comment by TECH7Fox** [comment]
All looks good to me. @bdherouville set it to ready when you want to merge.

**Comment by bdherouville** [comment]
Hi @TECH7Fox !
Done. I think the errors are not related to the content I made.

**Comment by TECH7Fox** [comment]
@bdherouville just need to change the version to `2.4.0` in `asterisk/config.yaml`

**Comment by bdherouville** [comment]
> @bdherouville just need to change the version to `2.4.0` in `asterisk/config.yaml`

Done, sorry for the closing by mistake. I intially thought I had to close this one to add the config.yaml commit.

**Comment by TECH7Fox** [comment]
Np. Thanks for the PR.

@felipecrs did you want to test the new version (#190) first?

File: asterisk/config.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: Asterisk
-version: 2.3.3
+version: 2.4.0
```
Comment (felipecrs) [inline asterisk/config.yaml:2]: You also need to update the CHANGELOG.md

**Review by felipecrs — CHANGES_REQUESTED** [review]

**Comment by felipecrs** [comment]
> @felipecrs did you want to test the new version (https://github.com/TECH7Fox/asterisk-hass-addons/pull/190) first?

Yes, but I can't test it right now. @bdherouville do you also have a VTO to test v2.3.4 before we merge this? Otherwise I can test it in a few hours.

File: asterisk/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
 
 # Changelog
 
+
+## 2.4.0
+
+- Add chan_sip on ports 52xx, but disabled in modules.conf for dahua vto compatibility (by @bdherouville)
```
Comment (felipecrs) [inline asterisk/CHANGELOG.md:8]:
```suggestion
- Add `chan_sip` on ports 52xx, but disabled by default in `modules.conf` for Dahua VTO compatibility (by @bdherouville)
```

**Comment by bdherouville** [comment]
> > @felipecrs did you want to test the new version (#190) first?
> 
> Yes, but I can't test it right now. @bdherouville do you also have a VTO to test v2.3.4 before we merge this? Otherwise I can test it in a few hours.

Yes, I have a VTO to test, do I have to upgrade my ha module ?

**Comment by bdherouville** [comment]
> Np. Thanks for the PR.
> 
> @felipecrs did you want to test the new version (#190) first?

Upgrade in progress...

**Comment by felipecrs** [comment]
Yes, you have to update the Asterisk add-on in your Home Assistant...

**Comment by felipecrs** [comment]
Just so you know, the new version upgrades Asterisk from 18 to 20. We want to test it because, who knows, maybe they fixed the Dahua issue on the new version.

**Comment by bdherouville** [comment]
> Just so you know, the new version upgrades Asterisk from 18 to 20. We want to test it because, who knows, maybe they fixed the Dahua issue on the new version.

I tried to convert/implement the PJSIP way of this SIP configuration [https://www.cctvforum.com/topic/48613-dahua-vto2000a-sip-firmware-and-asterisk/?do=findComment&comment=285319](https://www.cctvforum.com/topic/48613-dahua-vto2000a-sip-firmware-and-asterisk/?do=findComment&comment=285319) without success.

**Comment by felipecrs** [comment]
I guess you should use this? https://tech7fox.github.io/sip-hass-docs/docs/add-on/guides/doorbell#create-pjsip-contact

@TECH7Fox do you know anyone else that has a VTO to help verify that the extension in the link above actually works for a VTO?

**Comment by TECH7Fox** [comment]
In discord multiple people said that this was working:
```
[8000]
send_rpid=yes
send_pai=yes
from_domain=asterisk
rtp_symmetric=yes
force_rport=yes
rewrite_contact=yes
direct_media=no
context=default
disallow=all
allow=ulaw,alaw,speex,gsm,g726,g723,g722,opus
allow=h264,vp8,vp9
type = endpoint
context = default
;direct_media_method=invite
dtmf_mode=info
callerid="Doorbell" <8000>
;force_rport=no
aors= 8000
auth = auth8000

[8000]
type=aor
max_contacts=6
remove_existing=yes
remove_unavailable=yes
qualify_frequency=0

[auth8000]
type=auth
auth_type=userpass
password=8000         ; Set your password here
username=8000
```
https://discord.com/channels/912636682346582076/996826337727762463/1000507173392105532

**Comment by TECH7Fox** [comment]
But I think the one in the docs should also worked for VTO.

**Comment by bdherouville** [comment]
> In discord multiple people said that this was working:
> 
> ```
> [8000]
> send_rpid=yes
> send_pai=yes
> from_domain=asterisk
> rtp_symmetric=yes
> force_rport=yes
> rewrite_contact=yes
> direct_media=no
> context=default
> disallow=all
> allow=ulaw,alaw,speex,gsm,g726,g723,g722,opus
> allow=h264,vp8,vp9
> type = endpoint
> context = default
> ;direct_media_method=invite
> dtmf_mode=info
> callerid="Doorbell" <8000>
> ;force_rport=no
> aors= 8000
> auth = auth8000
> 
> [8000]
> type=aor
> max_contacts=6
> remove_existing=yes
> remove_unavailable=yes
> qualify_frequency=0
> 
> [auth8000]
> type=auth
> auth_type=userpass
> password=8000         ; Set your password here
> username=8000
> ```
> 
> https://discord.com/channels/912636682346582076/996826337727762463/1000507173392105532

Am I supposed to be able to call the doorbell with this configuration ?

**Comment by TECH7Fox** [comment]
I think so.

**Comment by bdherouville** [comment]
> I think so.

The link to discord channel redirect me to some landing page saying "no room here".

**Comment by felipecrs** [comment]
First join the group: https://discord.gg/aYmPHPrQ

**Comment by bdherouville** [comment]
> First join the group: https://discord.gg/aYmPHPrQ

Despite all my efforts I try to call the VTO but I have this log :

[Dec 20 22:22:30]   == Endpoint 8001 is now Reachable
[Dec 20 22:23:41]     -- Executing [8001@default:1] Progress("PJSIP/7777-00000000", "") in new stack
[Dec 20 22:23:41]     -- Executing [8001@default:2] Ringing("PJSIP/7777-00000000", "") in new stack
[Dec 20 22:23:41]     -- Executing [8001@default:3] Dial("PJSIP/7777-00000000", "PJSIP/8001,50,w") in new stack
[Dec 20 22:23:41]     -- Called PJSIP/8001
[Dec 20 22:23:41]     -- PJSIP/8001-00000001 is ringing
[Dec 20 22:24:12]     -- No one is available to answer at this time (1:0/0/0)
[Dec 20 22:24:12]     -- Auto fallthrough, channel 'PJSIP/7777-00000000' status is 'NOANSWER'

**Comment by felipecrs** [comment]
I don't understand much of Asterisk but it seems to me that you are calling 7777 instead of calling 8001?

**Comment by bdherouville** [comment]
> I don't understand much of Asterisk but it seems to me that you are calling 7777 instead of calling 8001? If that's the case, for the sake of testing, I think you should call the VTO directly (supposedly at 8001)

No, I am calling 8001 from 7777.
You can see it here : [Dec 20 22:23:41] -- PJSIP/8001-00000001 is ringing

**Comment by felipecrs** [comment]
Ok... time to test here too. I will boot up my VTO.

**Comment by felipecrs** [comment]
WOW! I can't believe it, everything is working!

audio + video through both incoming and outgoing call!

My doorbell is configured like this:

image

My extension is exactly like this: https://tech7fox.github.io/sip-hass-docs/docs/add-on/guides/doorbell#create-pjsip-contact

For me this is a Xmas Gift!

If someone else can verify it, we won't need chan_sip anymore.

**Comment by bdherouville** [comment]
Happy Christmas !
It works on my Dahua too!!!

Today I realized that I had a wifi repeater between my sip server and my doorbell.
I did a bit of tcpdump and the result showed that the doorbell was unable to send packets to the SIP server.

I commented the iptables modules that filtered the udp packets and everything workd immediately.

You will find below the tcpdump command that helped my to confirm the diagnostic.

tcpdump -u port 5060

20:54:26.437069 IP 111.222.333.444.5060 > 555.555.555.555.5060: SIP: INVITE sip:8001@555.555.555.555:5060 SIP/2.0
20:54:26.443400 IP 555.555.555.555.5060 > 111.222.333.444.5060: SIP: SIP/2.0 100 Trying
20:54:26.444062 IP 555.555.555.555.5060 > 111.222.333.444.5060: SIP: SIP/2.0 101 Dialog Establishment
20:54:26.451353 IP 555.555.555.555.5060 > 111.222.333.444.5060: SIP: SIP/2.0 180 Ringing
20:54:26.480204 IP 555.555.555.555.5060 > 111.222.333.444.5060: SIP: SIP/2.0 200 OK
20:54:28.474249 IP 555.555.555.555.5060 > 111.222.333.444.5060: SIP: SIP/2.0 200 OK
20:54:32.483895 IP 555.555.555.555.5060 > 111.222.333.444.5060: SIP: SIP/2.0 200 OK
20:54:36.494603 IP 555.555.555.555.5060 > 111.222.333.444.5060: SIP: SIP/2.0 200 OK
20:54:40.503624 IP 555.555.555.555.5060 > 111.222.333.444.5060: SIP: SIP/2.0 200 OK
20:54:44.514301 IP 555.555.555.555.5060 > 111.222.333.444.5060: SIP: SIP/2.0 200 OK
20:54:48.524142 IP 555.555.555.555.5060 > 111.222.333.444.5060: SIP: SIP/2.0 200 OK
20:54:52.534585 IP 555.555.555.555.5060 > 111.222.333.444.5060: SIP: SIP/2.0 200 OK

**Comment by TECH7Fox** [comment]
Ah, good you got it working.

So we probably don't have to add this anymore? The dahua VTO issue was the only reason to use chan_sip right?

**Comment by bdherouville** [comment]
Exact !
I bothered you for two uncomented lines in my openwrt wifi extender ... #facepalm

**Comment by felipecrs** [comment]
This is a great xmas gift from Asterisk team!

**Comment by TECH7Fox** [comment]
Reopening since there is still interest in chan_sip for older  devices.

**Comment by TECH7Fox** [comment]
@felipecrs do you know why it can't upload? 

image

**Comment by felipecrs** [comment]
It's possible that GitHub has changed their policy for GITHUB_TOKEN. I see some newer workflows have permissions: at the top level declaring the additional permissions it needs. Maybe additional permissions are now needed to publish to ghcr.

**Comment by felipecrs** [comment]
https://docs.github.com/en/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions#upgrading-a-workflow-that-accesses-ghcrio

**Comment by felipecrs** [comment]
https://github.com/tech7fox/asterisk-hass-addons/pkgs/container/amd64-addon-asterisk

**Comment by TECH7Fox** [comment]
I tried some things but I have no idea. Guessing a problem with the home assistant builder?
Anyways enabled bypass for now.

@felipecrs you requested some changes. Were those fixed?

**Comment by felipecrs** [comment]
I actually don't recall. But it's fine, if it's good for you it's good for me :)

**Comment by felipecrs** [comment]
After upgrading the add-on, my sip card can't connect to it anymore. The asterisk logs points to "Wrong password" errors.

Do I need to change anything in my configs? I am not trying to use chan_sip, and I didn't change anything in my configs, I just upgraded the add-on.

For now I restored the old version with the backup and it's working again.

**Comment by pergolafabio** [comment]
Should be no impact? Don't have issues here with pjsip (softphone)

Is chansip loaded? You can just use "noload" in the module file

**Comment by TECH7Fox** [comment]
Indeed try adding the `noload`. Or delete it to get the new default file again.

**Comment by felipecrs** [comment]
I'll try it. Thank you!

**Comment by felipecrs** [comment]
Isn't possible to have two extensions with the same number on both SIP and PJSIP? I thought it is, but if it is not, then we need to handle it indeed. Maybe the extensions in SIP should be generated from 200 instead of 100?

**Comment by pergolafabio** [comment]
i use both sip and pjsip at same time, should be no issue, just setup with different ports
5060 chansip
5160 pjsip

in dialplan you just do how you the endpoint is registered

like SIP/xxxx or PJSIP/xxx

**Comment by TECH7Fox** [comment]
Hmm, then I don't know why your endpoints would stop working...

**Comment by felipecrs** [comment]
I will do some more testing about it. But did someone who had a working setup try to upgrade to confirm if the issue isn't only me?

**Comment by pergolafabio** [comment]
well, i have no issues at all, but i dont  sip card, only softphones

have you tried the noload for chansip? then it should be the same as before

**Comment by felipecrs** [comment]
> have you tried the noload for chansip? then it should be the same as before

Will try, but I suppose the goal was to have both working at the same time, right?

**Comment by pergolafabio** [comment]
yeah, it works for me , i use both at same time with softphones 

i can call from pjsip extension to sip extension and visa versa

**Comment by felipecrs** [comment]
That did the trick. I resetted my asterisk files and now it works. Thanks!

**Comment by pergolafabio** [comment]
I think this issue is only for users that started with this addon when chan_sip was still present?

**Comment by felipecrs** [comment]
I don't think so, because I regularly reset my Asterisk files.

**Pull request merged**